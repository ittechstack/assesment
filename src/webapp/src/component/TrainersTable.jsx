import React, {useState} from 'react';
import 'react-toastify/dist/ReactToastify.css';
import {AddTrainerModal} from "./AddTrainerModal";

export const TrainersTable = props => {

    const {trainers, refreshWorkoutsOfTrainer, selectedTrainer, refreshTrainers} = props;

    const [showAddTrainer, setShowAddTrainer] = useState(false);

    return (
        <div className="col-sm">

            <AddTrainerModal
                show={showAddTrainer}
                onHide={() => {
                    setShowAddTrainer(false);
                    refreshTrainers();
                }}
            />

            <h3>Trainers</h3>
            <table className="table">
                <thead>
                <tr>
                    <th>Trainer Name</th>
                    <th></th>
                </tr>
                </thead>
                <tbody>
                {
                    trainers.map(trainer => {
                        return <tr key={trainer.id}
                                   style={selectedTrainer && selectedTrainer.id === trainer.id ? {backgroundColor: "rgba(0,0,0,.05)"} : {}}>
                            <td>{trainer.name}</td>
                            <td>
                                <button className="btn btn-primary btn-sm"
                                        onClick={() => refreshWorkoutsOfTrainer(trainer)}>
                                    Show Workouts
                                </button>
                            </td>
                        </tr>
                    })
                }
                </tbody>
            </table>
            <div className="row">
                <button className="btn btn-success" onClick={() => setShowAddTrainer(true)}>Add Trainer</button>
            </div>
        </div>
    )
};