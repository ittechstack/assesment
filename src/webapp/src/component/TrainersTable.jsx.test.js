import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import {TrainersTable} from "./TrainersTable";
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('TrainersTable component testing', () => {
    it('render correctly TrainersTable component', () => {
        const trainers = [{
            id: 1,
            name: 'Tom'
        }];
        const wrapper = renderer.create(<TrainersTable trainers={trainers}/>).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render header', () => {
        const trainers = [{
            id: 1,
            name: 'Tom'
        }];
        const wrapper = shallow(<TrainersTable trainers={trainers}/>);
        expect(wrapper.contains(<h3>Trainers</h3>)).toBe(true)
    });

    it('should render trainers table with 1 record', () => {
        const trainers = [{
            id: 1,
            name: 'Tom'
        }];
        const wrapper = shallow(<TrainersTable trainers={trainers}/>);

        const table = wrapper.find('table');
        expect(table).toHaveLength(1);

        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);

        const rows = tbody.find('tr');
        expect(rows).toHaveLength(trainers.length);
    });
});