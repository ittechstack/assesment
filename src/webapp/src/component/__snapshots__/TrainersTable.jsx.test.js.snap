// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainersTable component testing render correctly TrainersTable component 1`] = `
<div
  className="col-sm"
>
  <h3>
    Trainers
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Trainer Name
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        style={Object {}}
      >
        <td>
          Tom
        </td>
        <td>
          <button
            className="btn btn-primary btn-sm"
            onClick={[Function]}
          >
            Show Workouts
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="row"
  >
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Add Trainer
    </button>
  </div>
</div>
`;
